### PR TITLE
Describe how to build a Docker image with an example

### PR DESCRIPTION
**Pull request by wienans**
* Pull Request adds the ``docker tag`` command to the docs for container registry.

For Context i was using a own container registry the first time and also didn't pushed any docker images to docker.io so i wasn't aware of the necessity of the docker tag command. i only tried to follow the instructions in the page and got the error:

"An image does not exist locally with the tag: {domain}/{owner}/{image}:{tag}"

### CODE REVIEW

**Comment by KN4CK3R** [comment]
That's only related if you want to push an image you did not create in the first place. Maybe reword it to something like
> If you want to push an existing image, you must re-tag it.

And add it below the push command section because that should be part of a docker tutorial which the Gitea docs are not.

File: docs/content/usage/packages/container.en-us.md [hunk 1]
```diff
@@ -51,6 +51,12 @@ For example, these are all valid image names for the owner `testuser`:
 
 ## Push an image
 
+Before pushing an image it needs to be taged with the following command:
```
Comment (lunny) [inline docs/content/usage/packages/container.en-us.md]: This isn't always necessary if the image is only for this Gitea instance.

**Comment by wienans** [comment]
> That's only related if you want to push an image you did not create in the first place. Maybe reword it to something like
> 
> > If you want to push an existing image, you must re-tag it.
> 
> And add it below the push command section because that should be part of a docker tutorial which the Gitea docs are not.

Mhm okay. Actually I created that image myself. So not like I wanted to upload the Ubuntu latest  or something

**Comment by wienans** [comment]
@lunny moved the part under the full push section and added it as in case of an error.

**Review by lunny — APPROVED** [review]

**Comment by KN4CK3R** [comment]
I don't want to block the change but for me this is comparable to:

"We have a command `load <file>` which loads the file." Now the user executes `load /does/not/exist` and the command tells them "Can't find the file `/does/not/exist`". Would you expect that the documentation of `load` has the part: "If the command tells you the file can't be found, you must rename/move the file to what you passed as parameter to `load`." I would not expect that in the docs.

We already have a chapter https://docs.gitea.com/usage/packages/container#image-naming-convention which describes how the image name must be structured.

**Comment by wienans** [comment]
@KN4CK3R the confusing thing for me as a relative basic user of docker was that for me the tag was „latest“ and I didn’t get automatically that I need to write a new tag where the i needed tongue the domain image and tag into it.

My idea was to improve the handling for newer users. And I would have found that info in the doc helpful. But I also don’t mind if you don’t merge it.

**Comment by KN4CK3R** [comment]
I think you read the docs and may or may not have noticed the chapter I linked. Could we improve that block, so you don't run into the problem?
For example in that chapter I did not mention the `tag` (besides the ignored case). Would have adding `:{tag}` behind the examples helped you?

**Comment by KN4CK3R** [comment]
> Because I wasn’t aware that the „name“ of an image is also called tag?

No no no :sweat_smile: The name of an image is the part infront of the `:tag`. The `tag` is not mentioned in the docs because it's not relevant in the first place.

The "image name" is `{domain}/{owner}/{image}`. Maybe that's the confusing part in the docs because in the "image name" is an additional `{image}`? If you know Docker, that's not a problem, because you know `{image}` should be something like `apache` and the prefix is the url to the docker registry. But for new users that may not be clear. If we can find a better wording for the naming chapter I'm happy to change it.

**Comment by wienans** [comment]
> > Because I wasn’t aware that the „name“ of an image is also called tag?
> 
> No no no 😅 The name of an image is the part infront of the `:tag`. The `tag` is not mentioned in the docs because it's not relevant in the first place.
> 
> The "image name" is `{domain}/{owner}/{image}`. Maybe that's the confusing part in the docs because in the "image name" is an additional `{image}`? If you know Docker, that's not a problem, because you know `{image}` should be something like `apache` and the prefix is the url to the docker registry. But for new users that may not be clear. If we can find a better wording for the naming chapter I'm happy to change it.

Okay that at least is what I thought before my confusion started. But then I think the problem for starters is that docker cli itself is confusing in this parts. It returns can not find tag. In the push but actually it can not find the image, because the image does not even exist. Sure also the tag does not exist on that image. And then the command to „rename“ it is called `docker tag`.

Regarding doc improvement as soon as you know about all this the current docs are no problem. Without that for beginners or people like me which never got in touch with that specific part of docker, it might be not clear what exactly is meant with the naming convention and how to apply it.
In this case I still would add the specific command to the docs. But after the conversation with you I would put it under naming convention as a command to how to name your image like it is needed. But I think you are against adding this command at all to the doc, aren’t you?

**Comment by techknowlogick** [comment]
Perhaps a disclaimer that the hostname portion is required? And that otherwise docker will default to use docker hub?

**Review by a1012112796 — APPROVED** [review]

File: docs/content/usage/packages/container.en-us.md [hunk 2]
```diff
@@ -39,6 +39,12 @@ Images must follow this naming convention:
 
 `{registry}/{owner}/{image}`
 
+Name your image according to the convetion with the following command:
+
+```shell
+docker tag {image}:{tag} gitea.example.com/{owner}/{image}:{tag}
+```
+
```
Comment (techknowlogick) [inline docs/content/usage/packages/container.en-us.md]: What about:

```suggestion
When building your docker image, using the naming convention above, this looks like:
\```
docker build -t {registry}/{owner}/{image}:{tag} .
\```

where your registry is the domain of your gitea instance (e.g. gitea.example.com)
```

as it gives a full example of the build command showing an example to use the complete naming convention

Comment (KN4CK3R) [reply docs/content/usage/packages/container.en-us.md]: I like that 👍

Comment (wxiaoguang) [reply docs/content/usage/packages/container.en-us.md]: +1 . The key point is about the tag naming, but not the tag command.

But I am not sure whether the syntax of suggested change is right: why ```` \``` ```` there? Should it be ```` ```shell ````?

Comment (wxiaoguang) [reply docs/content/usage/packages/container.en-us.md]:
````suggestion
When building your docker image, using the naming convention above, this looks like:

```shell
# build an image with tag
docker build -t {registry}/{owner}/{image}:{tag} .
# name an existing image with tag
docker tag {some-existing-image}:{tag} {registry}/{owner}/{image}:{tag}
```

where your registry is the domain of your gitea instance (e.g. gitea.example.com)
````


(IMO if the "tag" command would be mentioned, maybe it could be this. Feel free to remove the "tag" command if it is redundant)

Comment (wienans) [reply docs/content/usage/packages/container.en-us.md]: 👍🏼

File: docs/content/usage/packages/container.en-us.md [hunk 3]
```diff
@@ -39,6 +39,16 @@ Images must follow this naming convention:
...
+When building your docker image, using the naming convention above, this looks like:
+
+```shell
+# build an image with tag
+docker build -t {registry}/{owner}/{image}:{tag} .
+# name an existing image with tag
+docker tag {some-existing-image}:{tag} {registry}/{owner}/{image}:{tag}
+```
+
+where your registry is the domain of your gitea instance (e.g. gitea.example.com)
```
Comment (KN4CK3R) [inline docs/content/usage/packages/container.en-us.md]:
```suggestion
where your registry is the domain of your gitea instance (e.g. gitea.example.com).
```

**Review by KN4CK3R — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Comment by KN4CK3R** [comment]
Do we need a backport for the docs?

**Comment by delvh** [comment]
Yes, as they are versioned.
